src/Controls/src/Core/Application.cs

```diff
@@ -262,6 +262,27 @@ void TriggerThemeChangedActual(AppThemeChangedEventArgs args)
 
 		public event EventHandler<ModalPushingEventArgs> ModalPushing;
```

Comment:
These events... Were they always here? Feels a bit weird now with Windows...

Reply:
I see you did that. Stunning!

Reply:
> These events... Were they always here?

HA yea I always forget they are here :-/ 

I mainly kept them here for backwards compatibility

Because users can currently do

`Application.Current.Modal events`

> Maybe this should be on page and then for compat, each window passes it up to the app as well?

Yea that's what the code does. Inside the Windows.impl.cs file I propagate these events up to Application 

https://github.com/dotnet/maui/pull/1417/files#diff-e95ac81676df57c3de0226dc120b0e527fbd52bc7fc506ddfde7cccf34a8f489R265

Reply:
You are so far ahead!